search_no_knn.go

```diff
@@ -60,25 +63,38 @@ type SearchRequest struct {
 	SearchAfter      []string          `json:"search_after"`
 	SearchBefore     []string          `json:"search_before"`
 
+	// metadata will be a  map that will be used
+	// in the second phase of any 2-phase search, to provide additional
+	// context to the second phase. This is useful in the case of index
+	// aliases where the first phase will gather the metadata from all
+	// the indexes in the alias, and the second phase will use that
+	// metadata to perform the actual search.
+	// The currently accepted map configuration is:
+	//
+	// "_knn_metadata_key": []*search.DocumentMatch
+
+	Metadata map[string]interface{} `json:"metadata,omitempty"`
```

Comment:
Is this needed in a non-knn setting?

Reply:
its generic to support extensibility in the future.
i can use this for sending in merged synonym definitions for example